### ISSUE TITLE
[Bug] Can not turn off logging to disk

### ISSUE THREAD

**Issue by kshivvy** [ray-project/ray#23451]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### Issue Severity

High: It blocks me to complete my task.

### What happened + What you expected to happen

I am running a ray.tune() experiment with num_samples > 50,000. My trial stops and runs into ""Disk quota exceeded" when running on a cluster, which I'm guessing because I've hit an inode limit from all the logs created in the ray_results/ directory.

How can I turn off ALL logging? I don't want any of it.  I've tried passing loggers=[NoopLogger] in tune.run(), but the directory still gets populated. 

Any tips?

### Versions / Dependencies

ray==1.11.0
Python 3.9.10
Windows (Linux on cluster)

### Reproduction script

```
from ray import tune
from ray.tune.logger import NoopLogger

def objective_fn(config):
    return 100

 analysis = tune.run(objective_fn,
                        config=search_space,
                        max_concurrent_trials=1,
                        resources_per_trial={"cpu": 1, "gpu":0},
                        num_samples=100000,

                        # Below does not turn off logging
                        loggers=[NoopLogger],
                        local_dir='./ray_results')
```

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by krfricke** [comment]
Hi @kshivvy, the problem is probably in the logging _callbacks_, which are automatically created. You can disable them with this environment variable:

```
export TUNE_DISABLE_AUTO_CALLBACK_LOGGERS=1
```

See also
https://docs.ray.io/en/latest/tune/api_docs/env.html

**Comment by kshivvy** [comment]
Hi @krfricke -

I've tried setting that like this:

`os.environ["TUNE_DISABLE_AUTO_CALLBACK_LOGGERS"] = "1"`

But the ray_results directory and objective_fn folders inside them still are created. Is there anything else I can do?

**Comment by krfricke** [comment]
The directories will always be created, but without the loggers it should not write anything to e.g. the results files there. Are the files still being written?

**Comment by krfricke** [comment]
Ah right, you also don't want to directories to be created because it's 50k. Let me take a look

**Comment by krfricke** [comment]
I'm afraid the log dirs will always be created and there is currently no great way around this. Tune generally assumes that there is a local directory to write artifacts to (e.g. checkpoints). Please let me know if the files are at least not being written to so that you don't run out of system memory.

**Comment by kshivvy** [comment]
Hi, 

The logs are no longer being written. I still run into Disk Quota issues, and from a cursory Google search it's not because I run out of space on the cluster, but because I hit the inode limit and can no longer create any more files or directories. 

I've tried decreasing the num_samples to 1000 but even then I run into this issue. 

Is there any workaround to this, even if it's complicated? In my objective function, what if I delete the ray_results/ folder? That way each trial will first delete the folder and then evaluate, so I'd likely never hit the inode limit.

**Comment by kshivvy** [comment]
I was running 50 experiments with 1000 samples each (50k samples total ), sorry for being unclear. This is the safe number of samples for me to run on my cluster. I found out that I was logging to tensorboard for each sample (it is a custom RL run), so I was producing an extra folder and file for every sample. I turned off that logging, and also added the Noop logger for raytune. After all this, the maximum number of samples I can run safely is ~20k.

I'm confident it was an inode issue and not space issue, since I had plenty of GB left. This happens on the Linux cluster, I'm ssh'ed directly. I haven't tested this behavior on my Windows machine (but I doubt it'd happen since my machine has a larger inode limit). 

Maybe a good feature would be to let a user prevent the logging from happening at all? I noticed some discussion here: https://github.com/ray-project/ray/issues/5048 but it looks like it was never implemented. 

For now though, I should be alright. Thanks for your help!
